### PR TITLE
[W5.11][W14-A1] Mukesh Gadupudi

### PR DESCRIPTION
**Pull request by shanmu9898**
Added two enhancements:
1) created a new command called "tagged" which listed every person based who has been assigned a certain tag.
Example : "tagged owesmoney" returns all the persons with the tag "owesmoney".

Updated test cases, documentation and code for the above.

2) Made the find command case insensitive.
Example: now it can find the person named John even when u type JOHN or JoHn.

Updated test cases, documentation and code for the above

### CODE REVIEW

**Comment by muruges95** [comment]
Hi @shanmu9898 everything looks good so far. Maybe, you could consider implementing case-insensitive search for tags as well seeing how you already have implemented it for the normal find operation. Furthermore, you could look to make the find by tag operation work similarly to the normal find, where you can search for multiple tags as well.

**Review by kexiaowen — COMMENTED** [review]
Good! One suggestion is that you can consider breaking up your changes to more commits instead of updating all code, tests and documentation at one go.

File: doc/UserGuide.md [hunk 1]
```diff
@@ -53,16 +53,25 @@ Examples:
 Shows a list of all persons in the address book.<br>
 Format: `list`
 
+### Listing all persons with a certain tag : `tagged`
+Shows a list of all persons in the address book with a certain tag. <br>
+Format: `tagged`
```
Comment (okkhoy) [inline doc/UserGuide.md:58]: You need to indicate that a tag name should be given as input.

File: src/seedu/addressbook/commands/TaggedCommand.java [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    public TaggedCommand(String tag) throws IllegalValueException {
+        numberOfTagsCalculator = tag.trim().split(" ");
+        this.tag = new Tag(tag.trim());
+    }
+
+    @Override
+    public CommandResult execute() {
+        if(numberOfTagsCalculator.length > 1){
+            return new CommandResult(String.format(MESSAGE_FAILURE));
+        }else{
```
Comment (okkhoy) [inline src/seedu/addressbook/commands/TaggedCommand.java:37]: Please take care of spacing between words and braces.

File: src/seedu/addressbook/parser/Parser.java [hunk 3]
```diff
@@ -11,17 +11,7 @@
...
-import seedu.addressbook.commands.Command;
-import seedu.addressbook.commands.DeleteCommand;
-import seedu.addressbook.commands.ExitCommand;
-import seedu.addressbook.commands.FindCommand;
-import seedu.addressbook.commands.HelpCommand;
-import seedu.addressbook.commands.IncorrectCommand;
-import seedu.addressbook.commands.ListCommand;
-import seedu.addressbook.commands.ViewAllCommand;
-import seedu.addressbook.commands.ViewCommand;
+import seedu.addressbook.commands.*;
```
Comment (okkhoy) [inline src/seedu/addressbook/parser/Parser.java:14]: Don't import everything using *; import the necessary classes explicitly

File: src/seedu/addressbook/parser/Parser.java [hunk 4]
```diff
@@ -137,6 +130,20 @@ private Command prepareAdd(String args) {
         }
     }
 
+    private Command prepareTagged(String args) {
+       // final Matcher matcher = TAGGED_DATA_ARGS_FORMAT.matcher(args.trim());
```
Comment (okkhoy) [inline src/seedu/addressbook/parser/Parser.java:134]: Remove unnecessary code completely; you can always recover it from previous versions RCS \#FTW

**Review by okkhoy — COMMENTED** [review]
Please take care of coding standards (particularly spacing between the names and braces).

**Comment by okkhoy** [comment]
@shanmu9898 
Some comments added. Please ack & close the PR after reading comments.

**Comment by shanmu9898** [comment]
Noted Sir! Thank you

**Pull request closed without merging**